src/ophys_etl/transforms/suite2p_wrapper.py

```diff
@@ -0,0 +1,219 @@
+import argschema
+import suite2p
+import pathlib
+import marshmallow as mm
+import h5py
+import tempfile
+from typing import List
+import datetime
+import shutil
+
+
+class Suite2PWrapperException(Exception):
+    pass
+
+
+class Suite2PWrapperSchema(argschema.ArgSchema):
+    # Options for running s2p. One list can be found at:
+    # https://github.com/MouseLand/suite2p/blob/master/suite2p/run_s2p.py
+
+    # s2p IO settings (file paths)
+    h5py = argschema.fields.InputFile(
+            required=True,
+            description="Path to 2P motion corrected hdf5 data path.")
+    h5py_key = argschema.fields.Str(
+            required=False,
+            missing="data",
+            default="data",
+            description="hdf5 file key for motion corrected dataset.")
+    # s2p registration settings
+    do_registration = argschema.fields.Int(
+            default=0,
+            description="0 skips registration, as we are doing our own.")
+    # s2p cell detection settings
+    roidetect = argschema.fields.Bool(
+            default=True,
+            description="Whether or not to run ROI extraction.")
+    sparse_mode = argschema.fields.Bool(
+            default=True,
+            description="Whether or not to run 'sparse mode'.")
+    diameter = argschema.fields.Int(
+            default=12,
+            description=("If not sparse_mode, use diameter (presumably "
+                         "in pixels) for filtering and extracting."))
+    spatial_scale = argschema.fields.Int(
+            default=0,
+            description=("0: multi-scale; 1: 6 pix; 2: 12 pix;"
+                         "3: 24 pix; 4: 48 pix"))
+    connected = argschema.fields.Bool(
+            default=True,
+            description="Whether to keep ROIs fully connected.")
+    nbinned = argschema.fields.Int(
+            required=False,
+            description="Max num of binned frames for cell detection.")
+    max_iterations = argschema.fields.Int(
+            default=20,
+            description="Max num iterations to detect cells.")
+    threshold_scaling = argschema.fields.Float(
+            default=0.75,
+            description=("Adjust automatically determined threshold by this "
+                         "scalar multiplier"))
+    max_overlap = argschema.fields.Float(
+            default=0.75,
+            description=("Cells with more overlap than this get removed "
```

Comment:
Is this a percent overlap? Might want to clarify in the description.

Reply:
I'll make it more clear above, but all of these are just copies of the Suite2P parameter names and their descriptions, courtesy of Nick in the `ophys_segmentation` repo. I don't think we should rename them.

Reply:
Understood, I didn't know they were copies, unfortunate their names are marginal.